### PR TITLE
Fb custom banner 5791

### PR DESCRIPTION
**Pull request by labkey-klum**

### CODE REVIEW

File: api/src/org/labkey/api/premium/PremiumService.java [hunk 1]
```diff
@@ -89,6 +92,11 @@ public boolean isRemoteREnabled()
         {
             return false;
         }
+
+        @Override
+        public ActionURL getConfCustomPageElements(Container c) {
+            return null;
```
Comment (labkey-klum) [inline api/src/org/labkey/api/premium/PremiumService.java]: brace formatting

File: api/src/org/labkey/api/settings/FooterProperties.java [hunk 2]
```diff
@@ -16,14 +16,32 @@
...
 {
-
     private final String FOOTER_CONFIGS = "FooterProperties";
     private final String SHOW_FOOTER_PROPERTY_NAME = "ShowFooter";
     private final String FOOTER_MODULE_PROPERTY_NAME = "FooterModule";
     private final String FILE_NAME = "_footer";
 
+    private Container _container;
+
+    // temporary
```
Comment (ilikepies101) [inline api/src/org/labkey/api/settings/FooterProperties.java]: Should we delete this function now? I don't think it's used anywhere and if it is, we can clean it up to use the constructor with the container.

File: api/src/org/labkey/api/settings/HeaderProperties.java [hunk 3]
```diff
@@ -25,6 +27,22 @@
     private final String HEADER_MODULE_PROPERTY_NAME = "HeaderModule";
     private final String FILE_NAME = "_header";
 
+    private Container _container;
+
+    @Deprecated
```
Comment (ilikepies101) [inline api/src/org/labkey/api/settings/HeaderProperties.java]: Same here for removing this constructor.

File: core/src/org/labkey/core/admin/lookAndFeelProperties.jsp [hunk 4]
```diff
@@ -131,9 +131,11 @@
     <td class="labkey-form-label">Support email (shown to users if they don't have permission<br/>to see a page, or are having trouble logging in)</td>
     <td style="vertical-align: top;"><input type="text" name="supportEmail" size="50" value="<%= h(laf.getSupportEmail()) %>"></td>
 </tr>
-<tr>
-    <td>&nbsp;</td>
-</tr>
+<% if (bean.getCustomPageElementLink(c) != null) { %>
```
Comment (ilikepies101) [inline core/src/org/labkey/core/admin/lookAndFeelProperties.jsp]: Move curly brace to keep styling consistent

File: api/src/org/labkey/api/settings/FooterProperties.java (hunk 2 above)
Comment (labkey-klum) [reply api/src/org/labkey/api/settings/FooterProperties.java]: yes, I'll go ahead and remove those deprecated constructors

**Pull request closed without merging**